ast.c

```diff
@@ -884,6 +921,16 @@ static void ast_to_zval(zval *zv, zend_ast *ast, ast_state_info_t *state) {
 		ast_update_property_long(&prop_group_zval, AST_STR(str_flags), 0, AST_CACHE_SLOT_FLAGS);
 	}
 #endif
+#if PHP_VERSION_ID < 80000
+	if (ast->kind == ZEND_AST_CLASS_CONST_DECL && state->version >= 80) {
+		zval const_decl_zval = *zv;
```

Comment:
ZVAL_COPY_VALUE

Reply:
This was based on the code for property groups. The total reference counts to the ast\Node didn't change, so I think ZVAL_COPY_VALUE would lead to a leak - it's being replaced in place for the ast_create_virtual_node_ex call

Reply:
ZVAL_COPY_VALUE does not change the refcount, it's basically the same as assigning the value, with the difference that it does not touch u2. It makes not practical difference here, just a matter of general policy (which is to never perform direct zval assignments).